creator-node/test/users.js

```diff
@@ -48,6 +52,117 @@ describe('test Users', function () {
       .expect(200)
   })
 
+  it('fail to get challenge without wallet address', function (done) {
+    request(app)
+      .get('/users/login/challenge')
+      .expect(400, done)
+  })
+
+  it('get challenge with wallet address', async function () {
+    await createStarterCNodeUser()
+    await request(app)
+      .get('/users/login/challenge')
+      .query({ walletPublicKey: testEthereumConstants.pubKey })
+      .expect(200)
+  })
+
+  it('fail using POST challenge route with missing body keys', async function () {
+    await request(app)
+      .post('/users/login/challenge')
+      .send({ })
+      .expect(400)
+  })
+
+  it('fail using POST challenge route with invalid data and signature', async function () {
+    await request(app)
+      .post('/users/login/challenge')
+      .send({ data: 'data', signature: 'signature' })
+      .expect(400)
+  })
+
+  it('fail using POST challenge route with no cnode user', async function () {
+    let challengeResp
+    await request(app)
+      .get('/users/login/challenge')
+      .query({ walletPublicKey: testEthereumConstants.pubKey })
+      .expect(200)
+      .then(response => {
+        challengeResp = response.body
+      })
+
+    const signature = sigUtil.personalSign(Buffer.from(testEthereumConstants.privKeyHex, 'hex'), { data: challengeResp.challenge })
+
+    await request(app)
+      .post('/users/login/challenge')
+      .send({ data: challengeResp.challenge, signature })
+      .expect(400)
+  })
+
+  it('fail using POST challenge route with challenge key not present in redis', async function () {
+    this.timeout(200000) // set timeout for mocha test to accomodate for setTimeout call
+    let challengeResp
+    await createStarterCNodeUser()
+    await request(app)
+      .get('/users/login/challenge')
+      .query({ walletPublicKey: testEthereumConstants.pubKey.toUpperCase() })
+      .expect(200)
+      .then(response => {
+        challengeResp = response.body
+      })
+
+    await wait(130000)// wait for redis cache to expire; current ttl is 120s
```

Comment:
i don't think this is a reasonable during a test. as a proxy for this, i think maybe we could pass in an incorrect public key into the GET /users/login/challenge route and then that would fail to retrieve the value from redis when you sign and send the correct signature back